mock-server/mock_server.py

```diff
@@ -143,13 +145,21 @@ def tx_submit(params, *args):
             balances[params["from"]] = sender_balance - amount
             balances[params["to"]] = balances.get(params["to"], 0) + amount
         address = params["from"]
+        # Generate a unique TEST tx_hash
+        tx_hash = f'{address[2:]}{int(params["nonce"]):05x}00000000000deadbeef'
+        transactions[tx_hash] = {
+            'to': params["to"],
+            'from': address,
+            'amount': params["amount"],
+            'created_at': datetime.now().isoformat("T")
+        }
 
     elif params["type"] == "ChangePubKey":
         address = params["account"]
-
-    if address:
         # Generate a unique TEST tx_hash
         tx_hash = f'{address[2:]}{int(params["nonce"]):05x}00000000000deadbeef'
```

Comment:
My only concern is that we previously checked for the `address` before we slice `address[2:]`.
Now the test is moved a few lines down...

Reply:
i think this is safe due to the `address` beeing set on the line above the slice.

The only case where address is not set is when type=Withdraw